File: leaderboard.go
package main

type LeaderboardEntry struct {
	Id    int    `json:"id"`
	Name  string `json:"name"`
	Moves int    `json:"moves"`
	Hash  string `json:"hash"`
}

type Leaderboard struct {
	Entries []LeaderboardEntry `json:"entries"`
}

File: static/js/app/collections/leaderboard.js
define(function(require) {

  var Backbone = require('backbone');
  var LeaderboardEntry = require('app/models/leaderboard_entry');

  var Leaderboard = Backbone.Collection.extend({

    model: LeaderboardEntry,

    initialize: function() {
      this.page = 1;
    },

    url: function() {
      return '/api/v1/leaderboards?page=' + this.page;
    },

    next: function() {
      this.page += 1;
      this.fetch({ remove: false });
    },

    parse: function(response) {
      return response.entries;
    }

  });

  return Leaderboard;

});
